timemachine/cpp/src/wrap_kernels.cpp

```diff
@@ -180,6 +180,84 @@ void declare_context(py::module &m) {
         ctxt.get_x_t(buffer.mutable_data());
         return buffer;
     })
+    .def("multiple_steps_U", [](timemachine::Context &ctxt,
+        const double lambda,
+        const int n_steps,
+        const py::array_t<double, py::array::c_style> &lambda_windows,
+        int store_u_interval,
+        int store_x_interval) -> py::tuple {
+   
+        std::vector<double> vec_lambda_windows(lambda_windows.size());
+        std::memcpy(vec_lambda_windows.data(), lambda_windows.data(), vec_lambda_windows.size()*sizeof(double));
+
+        int u_interval = (store_u_interval <= 0) ? lambda_windows.size() : store_u_interval;
+        int x_interval = (store_x_interval <= 0) ? lambda_windows.size() : store_x_interval;
+
+        std::array<std::vector<double>, 3> result = ctxt.multiple_steps_U(
+            lambda,
+            n_steps,
+            vec_lambda_windows,
+            u_interval,
+            x_interval
+        );
+
+        int UF = result[0].size()/lambda_windows.size();
+        int UW = lambda_windows.size();
+
+        py::array_t<double, py::array::c_style> out_u_buffer({UF, UW});
+        std::memcpy(out_u_buffer.mutable_data(), result[0].data(), result[0].size()*sizeof(double));
+
+        int N = ctxt.num_atoms();
+        int D = 3;
+        int F = (n_steps + x_interval - 1) / x_interval;
+        py::array_t<double, py::array::c_style> out_x_buffer({F, N, D});
+        std::memcpy(out_x_buffer.mutable_data(), result[1].data(), result[1].size()*sizeof(double));
+
+        py::array_t<double, py::array::c_style> box_buffer({F, D, D});
+        std::memcpy(box_buffer.mutable_data(), result[2].data(), result[2].size()*sizeof(double));
+
+        return py::make_tuple(out_u_buffer, out_x_buffer, box_buffer);
+    }, R"pbdoc(
+        Compute energies across multiple lambda windows while simulating
+        at a single fixed lambda window.
+        
+        Let lambda_windows have shape [K].
+        F = ceil(n_steps/store_u_interval).
+        
+        The returned U matrix has F rows and K columns.
+
+        Parameters
+        ----------
+        lambda: float
+            Lambda window we run the simulation at
+
+        n_steps: int
+            Number of steps to run.
+
+        lambda_windows: np.array, of shape K
+            Lambda values to evaluate energies at
+
+        store_u_interval: int
+            How often we store the energies
+
+        store_x_interval: int
+            How often we store the frames
+
+        Returns
+        -------
+        3-tuple of energies, coordinates, boxes
+            Energies have shape (ceil(n_steps/store_u_interval), K)
+            Coordinates have shape (ceil(n_steps/store_x_interval), N, 3)
+            Boxes have shape (ceil(n_steps/store_x_interval), 3, 3)
```

Comment:
Nit:
```suggestion
            Energies have shape (F, K)
            Coordinates have shape (F, N, 3)
            Boxes have shape (F, 3, 3)
```
since `F = ceil(n_steps/store_u_interval)` is introduced and used earlier in this docstring